tests/test_edge_cases.py

```diff
@@ -957,15 +957,48 @@ def __init__(self, **data) -> None:
     Foobar(x=1)
 
 
-def test_ignored_type():
-    def foobar():
+def test_type_on_annotation():
+    class FooBar:
         pass
 
     class Model(BaseModel):
-        a: int = foobar
-        b: int
+        a: int = int
+        b: Type[int]
+        c: Type[int] = int
+        d: FooBar = FooBar
+        e: Type[FooBar]
+        f: Type[FooBar] = FooBar
+
+    assert Model.__fields__.keys() == {'b', 'c', 'e', 'f'}
+
+    class Parent:
```

Comment:
From here on I think this should be a separate test (you'll need to define `Model` with the fields required).